Enable Develocity build cache by default and document it better

gsmet opened the pull request.

gsmet committed:
Enable Develocity build cache by default and document it better

maxandersen commented on CONTRIBUTING.md:
```diff
@@ -531,21 +530,32 @@ CI is using a slightly different GIB config than locally:
 
-If you have a need or interest to report build times, you will need to get an API key for the GE instance. It is mainly relevant for those working on optimizing the Quarkus build. Ping on quarkus-dev mailing list or on Zulip if you need one.
+If you have a need or interest to share your build scans and use the build cache, you will need to get an account for the Develocity instance.
+It is only relevant for members of the Quarkus team and you should contact either Max Andersen or Guillaume Smet to set up your account.
```
any contributor can get it so no real need to put request in private is there?

maxandersen commented on .mvn/gradle-enterprise.xml:
```diff
@@ -26,7 +26,7 @@
     <buildCache>
         <local>
-            <enabled>#{env['GRADLE_LOCAL_BUILD_CACHE'] != null and env['RELEASE_GITHUB_TOKEN'] == null and properties['no-build-cache'] == null}</enabled>
+            <enabled>#{env['RELEASE_GITHUB_TOKEN'] == null and properties['no-build-cache'] == null}</enabled>
```
shouldn't this not only be enabled if access key available?

gsmet replied:
Ah ah funnily enough, I thought about that during the meeting and already pushed something related.

This is the local cache so we don't care but I added the check for the remote cache just below though (but my understanding is that it's handled anyway).

gsmet committed:
Add a Gradle profiler scenario and some instructions

gsmet committed:
Update quarkus-project-develocity-extension to 1.0.1

gsmet committed:
Add more documentation

gsmet committed:
Make docs more efficient when using -DskipDocs

gsmet committed:
Allow caching the testCompile goal for extensions

Only the compile goal is problematic for extensions.

gsmet committed:
Move Kotlin build cache configuration to the extension

gsmet committed:
Enforce a static outputTimestamp for reproducible builds

And update it before releasing.
This is important for the build cache as we are generating checksums of
zips which are dependent on the timestamp.

gsmet committed:
Build cache - Move XML config to the Maven extension

gsmet committed:
Reenable the remote cache

gsmet committed:
Build cache - Move the Groovy code to the Maven extension
